sphinx/util/nodes.py

```diff
@@ -33,7 +34,10 @@
 caption_ref_re = explicit_title_re  # b/w compat alias
 
 
-class NodeMatcher:
+T = TypeVar("T", bound=Node)
```

Comment:
Maybe use `N` or `NT` for "NodeType" instead of `T`?

Reply:
done